pkg/model/api/v1/datasource.go

```diff
@@ -20,6 +20,23 @@ import (
 	"github.com/perses/perses/pkg/model/api/v1/common"
 )
 
+func FilterDatasource[T DatasourceInterface](kind string, defaultDTS *bool, list []T) []T {
+	result := make([]T, 0, len(list))
+	for _, d := range list {
+		keep := true
+		if len(kind) > 0 && kind != d.GetSpec().Plugin.Kind {
+			keep = false
+		}
```

Comment:
else if instead ? no need to go through the second one if keep is already set to `false`

Reply:
I could even merge the conditions together.
Like 

```go
              if (len(kind) > 0 && kind != d.GetSpec().Plugin.Kind) && 
                  defaultDTS != nil && *defaultDTS != d.GetSpec().Default {
			keep = false
		}
```

WDYT ?

Reply:
What about

```go
for _, d := range list {
    if (len(kind) > 0 && kind != d.GetSpec().Plugin.Kind) || (defaultDTS != nil && *defaultDTS != d.GetSpec().Default) {
        continue
    }
    result = append(result, d)
}
```

?

Reply:
even better. Thank you !